### PR TITLE
Don't stop if a variant can't be synthesized

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tools/main.toit [hunk 1]
```diff
@@ -121,14 +123,24 @@ variant_synthesize parsed/cli.Parsed --ui/cli.Ui:
...
+    exception := catch:
+      variant_synthesize
+          --variant_path="$variants_root/$variant"
+          --toit_root=toit_root
+          --output="$output_root/$variant"
+          --build_path="$build_root/$variant"
+          --sdk_path=sdk_path
+          --ui=ui
+    if exception:
+      ui.print "Failed to synthesize variant '$variant': $exception"
```
Comment (kasperl) [inline tools/main.toit:138]: Terminate with .?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -158,6 +159,11 @@ jobs:
           ccache -s
           for variant in $VARIANTS; do
             mkdir -p build/$variant
+            if [ -e synthesized/$variant/failed ]; then
+              echo "Skipping $variant to due failed synthesis"
```
Comment (kasperl) [inline .github/workflows/build.yml:163]: Terminate with .?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/main.toit (hunk 1 above)
Comment (floitsch) [reply tools/main.toit]: done

File: .github/workflows/build.yml (hunk 2 above)
Comment (floitsch) [reply .github/workflows/build.yml]: done.

**Pull request merged**